test/export/InstanceExporter.test.ts

```diff
@@ -8717,6 +8721,213 @@ describe('InstanceExporter', () => {
+        carProfile = new Profile('CarProfile');
+        carProfile.parent = 'Car';
+        doc.profiles.set(carProfile.name, carProfile);
+        carInstance = new Instance('CarInstance');
+        carInstance.instanceOf = 'Car';
+        doc.instances.set(carInstance.name, carInstance);
+        carProfileInstance = new Instance('CarProfileInstance');
+        carProfileInstance.instanceOf = 'CarProfile';
+        doc.instances.set(carProfileInstance.name, carProfileInstance);
+      });
+
+      afterEach(() => {
+        // None of the test expect warnings or errors. All should be clean.
+        expect(loggerSpy.getAllLogs('warn')).toHaveLength(0);
+        //expect(loggerSpy.getAllLogs('error')).toHaveLength(0);
+      });
+
+      // Setting resourceType
+
+      it('should set resourceType to the logical type we are making an instance of', () => {
+        const exported = exportInstance(carInstance);
+        expect(exported.resourceType).toBe(
+          'http://hl7.org/fhir/us/minimal/StructureDefinition/Car'
+        );
+      });
+
+      it('should set resourceType to the logical type for the profile of a logical we are making an instance of', () => {
+        const exported = exportInstance(carProfileInstance);
+        expect(exported.resourceType).toBe(
+          'http://hl7.org/fhir/us/minimal/StructureDefinition/Car'
+        );
+      });
+
+      // Setting meta.profile
+
+      it('should not set meta.profile when we are making an instance of a logical', () => {
+        const exported = exportInstance(carInstance);
+        expect(exported.meta).toBeUndefined();
+      });
+
+      it('should not set meta.profile when we are making an instance of a logical even when it has meta', () => {
+        const metaRule = new AddElementRule('meta');
+        metaRule.min = 0;
+        metaRule.max = '1';
+        metaRule.types = [{ type: 'Meta' }];
+        carLogical.rules.push(metaRule);
+        const exported = exportInstance(carInstance);
+        expect(exported.meta).toBeUndefined();
+      });
+
+      it('should not set meta.profile when we are making an instance of a profile of logical that has no meta', () => {
+        const exported = exportInstance(carProfileInstance);
+        expect(exported.meta).toBeUndefined();
+      });
+
+      it('should set meta.profile to the defining profile URL we are making an instance of logical (for profile of logical that has meta)', () => {
+        const metaRule = new AddElementRule('meta');
+        metaRule.min = 0;
+        metaRule.max = '1';
+        metaRule.types = [{ type: 'Meta' }];
+        carLogical.rules.push(metaRule);
+        const exported = exportInstance(carProfileInstance);
+        expect(exported.meta).toEqual({
+          profile: ['http://hl7.org/fhir/us/minimal/StructureDefinition/CarProfile']
+        });
+      });
+
+      it('should not set meta.profile when we are making an instance of a profile of a logical with >1 meta', () => {
```

Comment:
Sorry for the likely silly question, but why is it that we only want to set the `meta.profile` when the base cardinality has max 1, and not when the max is > 1 or has been constrained to 1?

Reply:
Three reasons:

1) All FHIR resources have `0..1` meta -- so we'll only do this for things that conform to that shape -- because we know what's generally expected.
2) If there are `0..*` metas (in a logical or custom resource), we have no idea what that means and how it's intended to be used.
3) From a practical standpoint, we just don't want to have to support array serialization of `meta[]`.

Reply:
Ah, that all makes a lot of sense. Thanks for the explanation!